src/main/java/seedu/address/model/cheese/Cheese.java

```diff
@@ -69,6 +71,25 @@ public boolean getAssignStatus() {
         return isAssigned;
     }
 
+    /**
+     * Checks whether the given parameters are valid for an order
+     */
+    public static void checkCheeseArguments(CheeseType cheeseType, ManufactureDate manufactureDate,
+                                            MaturityDate maturityDate, ExpiryDate expiryDate,
+                                            CheeseId cheeseId, boolean isAssigned) {
+        if (maturityDate != null) {
+            checkArgument(maturityDate.isAfter(manufactureDate),
+                "The maturity date of the cheese should be after the manufacture date.");
+        }
+
```

Comment:
I think we can add in a case to handle when the user keys in an expiry date but not a maturity date when adding a cheese.

Anyway, since we are at this topic: I was thinking, we may want to consider dropping certain attributes from the models. For example only, since we have only 1 week left to add features, I think we may not have the time to implement features that deal with the cheeses' manufacture date, maturity date, etc.

We can always leave the clean up to v1.4, so it's just something to think about now? Should discuss more during Saturday's meeting.

Reply:
Which would you suggest to drop? If we want, we can just keep expiry and manufacture date (then I can add one more case in the AddressBook to ensure that the Cheese's manufacture date must be after the order's order date)

Reply:
This one you and Li Quan can decide? or you want to wait until Saturday to discuss also can.

And then, actually a cheese manufacture date don't necessarily have to be after an order's order date right? Are you referring to the order's completed date instead? If so then sounds good!

Reply:
Will discuss on Saturday